src/ui/server/controllers/insurance/business/company-details/validation/company-details/rules/tradingName.test.ts

```diff
@@ -0,0 +1,69 @@
+import tradingName from './tradingName';
+import { ERROR_MESSAGES } from '../../../../../../../content-strings';
+import { FIELD_IDS } from '../../../../../../../constants';
+import generateValidationErrors from '../../../../../../../helpers/validation';
+import { RequestBody } from '../../../../../../../../types';
+
+const {
+  YOUR_COMPANY: { TRADING_NAME },
+} = FIELD_IDS.INSURANCE.EXPORTER_BUSINESS;
+
+const { EXPORTER_BUSINESS } = ERROR_MESSAGES.INSURANCE;
+
+describe('controllers/insurance/business/company-details/validation/company-details/rules/tradingName', () => {
+  const mockErrors = {
+    summary: [],
+    errorList: {},
+  };
+
+  const mockBody = {
+    TRADING_NAME: '',
+  } as RequestBody;
+
+  it('should return a validation error when tradingName is an empty string', () => {
+    const result = tradingName(mockBody, mockErrors);
+
+    const errorMessage = EXPORTER_BUSINESS[TRADING_NAME].IS_EMPTY;
+    const expected = generateValidationErrors(TRADING_NAME, errorMessage, mockErrors);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should return a validation error when tradingName is null string', () => {
+    mockBody[TRADING_NAME] = null;
+
+    const result = tradingName(mockBody, mockErrors);
+
+    const errorMessage = EXPORTER_BUSINESS[TRADING_NAME].IS_EMPTY;
+    const expected = generateValidationErrors(TRADING_NAME, errorMessage, mockErrors);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should return a validation error when tradingName is not in the body', () => {
+    delete mockBody[TRADING_NAME];
+
+    const result = tradingName(mockBody, mockErrors);
+
+    const errorMessage = EXPORTER_BUSINESS[TRADING_NAME].IS_EMPTY;
```

Comment:
Could we extract this? Defined a few times in the test

Reply:
done